src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8957,55 +8916,52 @@ private TryStatementSyntax ParseTryStatement(SyntaxList<AttributeListSyntax> att
                 _termState = saveTerm;
             }
 
-            var catches = default(SyntaxListBuilder<CatchClauseSyntax>);
-            FinallyClauseSyntax @finally = null;
+            SyntaxListBuilder<CatchClauseSyntax> catchClauses = default;
+            FinallyClauseSyntax finallyClause = null;
             try
             {
-                bool hasEnd = false;
-
                 if (this.CurrentToken.Kind == SyntaxKind.CatchKeyword)
                 {
-                    hasEnd = true;
-                    catches = _pool.Allocate<CatchClauseSyntax>();
+                    catchClauses = _pool.Allocate<CatchClauseSyntax>();
                     while (this.CurrentToken.Kind == SyntaxKind.CatchKeyword)
-                    {
-                        catches.Add(this.ParseCatchClause());
-                    }
+                        catchClauses.Add(this.ParseCatchClause());
                 }
```

Comment:
This local function doesn't feel necessary. Other code does: `SyntaxFactory.MissingToken(SyntaxKind.XYZ)`